internal/config/config.go

```diff
@@ -65,13 +66,20 @@ func RegisterConfigFile() error {
 
 func GetConfig() *Config {
 	config := &Config{
-		Version:        viperInstance.GetString(VersionConfigKey),
-		Log:            getLog(),
-		ProcessMonitor: getProcessMonitor(),
-		DataplaneAPI:   getDataplaneAPI(),
-		Client:         getClient(),
+		Version:            viperInstance.GetString(VersionConfigKey),
+		Log:                getLog(),
+		ProcessMonitor:     getProcessMonitor(),
+		DataplaneAPI:       getDataplaneAPI(),
+		Client:             getClient(),
+		ConfigDir:          getConfigDir(),
+		AllowedDirectories: []string{},
 	}
 
+	for _, dir := range strings.Split(config.ConfigDir, ":") {
+		if dir != "" {
+			config.AllowedDirectories = append(config.AllowedDirectories, dir)
```

Comment:
probably should validate that the directory path is an absolute path before adding it to the array. If its not it shouldn't be added and a warning message should be logged.